src/main/java/org/mule/tools/devkit/sonar/checks/maven/DevkitLatestVersionCheck.java

```diff
@@ -0,0 +1,121 @@
+package org.mule.tools.devkit.sonar.checks.maven;
+
+
+
+import com.google.common.base.Splitter;
+import com.google.common.collect.Lists;
+import org.apache.maven.project.MavenProject;
+import org.mule.tools.devkit.sonar.checks.ConnectorIssue;
+
+import javax.xml.stream.XMLInputFactory;
+import javax.xml.stream.XMLStreamException;
+import javax.xml.stream.XMLStreamReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.util.List;
+
+public class DevkitLatestVersionCheck implements MavenCheck {
+
+    public static final String KEY = "devkit-latest-version";
+
+    @Override
+    public Iterable<ConnectorIssue> analyze(MavenProject mavenProject) {
+        final List<ConnectorIssue> issues = Lists.newArrayList();
+        Version currentVersion = new Version(mavenProject.getModel().getParent().getVersion());
+        try{
+            Version latestVersion = getLatestVersion();
+            if(currentVersion.compareTo(latestVersion) == -1){
+                logAndRaiseIssue(issues, String.format("Current Devkit version: '%s' is not up to date. Please download the latest version if possible: '%s'", mavenProject.getModel().getParent().getVersion(),latestVersion));
+            }
+        }catch(IOException|XMLStreamException e){
+            logAndRaiseIssue(issues, "Error retrieving the XML");
+        }
+        return issues;
+    }
+
+    public Version getLatestVersion() throws IOException, XMLStreamException {
```

Comment:
The XMLStreamReader implementation seems too complex here if your only goal is to obtain the list of <version>. I'd rather go for a **DOM approach** instead. This way, you can  read the XML as a tree and fetch the node you want (and its childs) without going through the entire structure. More importantly, it will let you get rid of those suspicious *do / while* loops.